app/models/job/dispatcher.rb

```diff
@@ -19,20 +19,26 @@ def dispatch
       raise NotImplementedError, _("Must be implemented in a subclass")
     end
 
-    def pending_jobs
-      zone = MiqServer.my_zone
+    def zone
+      @zone ||= Zone.find_by(:name => zone_name)
+    end
 
+    def zone_name
+      @zone_name ||= MiqServer.my_zone
+    end
+
+    def pending_jobs
       job_class.order(:id)
                .where(:state           => "waiting_to_start")
                .where(:dispatch_status => "pending")
-               .where("zone is null or zone = ?", zone)
+               .where("zone is null or zone = ?", zone_name)
```

Comment:
Looks good!  I've been wanting to add or rename `my_zone` to `my_zone_name` because it's so confusing.  I need to verify it each time I come back to it to make sure it's the zone object or the zone.name.

Reply:
:100: agree